can-construct_test.js

```diff
@@ -159,3 +159,44 @@ test("setup called with original arguments", function(){
 
 	Construct.extend(o1, o2);
 });
+
+test("legacy namespace strings (A.B.C) accepted", function() {
+
+	var Type = Construct.extend("Foo.Bar.Baz");
+
+	ok(new Type() instanceof Construct, "No unexpected behavior in the prototype chain");
+	equal(Type.name, "Foo_Bar_Baz", "Name becomes underscored");
```

Comment:
There's a couple things that maybe aren't obvious from the test as written:

* The "name" property is set by the VM and immutable.  What we do in production currently, sets the name to "Constructor" in Chrome and the empty string in Firefox.
* Because of this, we could never have a name property with anything but valid variable names.  The VM simply won't allow it.
* shortName is what we should be focusing on for dev/prod consistency, since that's what we have more control over.

BTW, I would totally recommend not doing `eval()` at all, since we could get the same results (with name and all) doing `Constructor = new Function("return function " + constructorName + "() {return init.apply(this, arguments); }")();` -- would this make it so that we could do this safely/sanely in prod?  Would we even want to do that?